### ISSUE TITLE
From: when Bounce

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#362]
**Originally by _johnblock@storeshop.com_ on 2012-12-29 18:38:32 +0100**

___
On Bounce, From: has always been the original sender`s From:

Now it overwrites with users email.

Please change back to original sender`s From: or make it an option.

**Comment by jens-maus** [comment]
**Originally on 2012-12-29 18:50:36 +0100**

___
Before considering your request, can you please elaborate why you feel the original senders' From: should be added when you use the Bounce functionality? For which purpose do you use this function?

**Comment by jens-maus** [comment]
**Originally by _johnblock@storeshop.com_ on 2013-01-01 18:16:54 +0100**

___
It is a core feature used for keeping senders details intact.
I use it to bounce emails for others to deal with while leaving me out of it eg misdirected emails. Redirecting enquiries to auto responders which use From: to address sender. Sending on emails to warehouse where they need to know who the order is from. Checking sent drawer to see when orders sent. Returning emails like newsletters back to mailbox for later reading or dealing with.
Ideally From would also be optionally free to put some thing else in as do write emails for other people, but can use the identity to create a temporary sepparate identity.
Have gone back to 2.7 to get back a functional bounce.

**Comment by jens-maus** [comment]
**Originally on 2013-01-01 18:36:25 +0100**

___
After reading your request and thinking about the matter again, I agree that we should change the current behaviour if the 'bounce' functionality to comply with common standards. See, e.g. http://mailredirect.mozdev.org/ for a similar functionality provided for thunderbird. However, we should do the following to be fully compliant:

1. rename the functionality to 'Mail Redirect' because Bounce is not clear enough .
2. add support for all the additional 'Mail-Resent' header entries mentioned in the rfc2822 standard and link above.

If all of the above is fullfilled I think fixing the bounce functionality like you expecting it would be possible.

**Comment by jens-maus** [comment]
**Originally by _johnblock@storeshop.com_ on 2013-01-02 10:45:28 +0100**

___
[http://yam.ch/wiki/FAQ/Using%20YAM#WhatsthepurposeoftheBounceoptioninYAMandcanIuseittoe.g.bounceSPAMmailback]

FAQ 4:3 Bounce has been included as you describe since version 2.3

And called Bounce

Please if you have to change the name make it Bounce Redirect

The Amiga B combination is used for Bounce Amiga - M (Mail Redirect) Amiga - R (Redirect) are in use
It has been Bounce since 2.3 back in 2001 without problem and would be known for 12 years as Bounce by people who use it
rfc2822 Feature of mail redirecting is also known as bouncing.

**Comment by jens-maus** [comment]
**Originally on 2013-01-02 10:55:21 +0100**

___
Thanks for the hint. It has benn long ago since I enhanced the bounce functionality in YAM. With the new identity management I might have introduced the current missbehaviour. Nevertheless, just because the feature has been called Bounce since 2001 doesn't make it more correct (Bounce gives the impression that the festure can be used to sent mail back to the sender easily). Things change over time and so the naming, but I will take care not to change the hotkey if possible. In addition, I have to amend the FAQ entry to add some comments on implications on having redirected mail identified as SPAM today since you are usually not allowed to sent mail with a foreign From:.

Nevertheless, the ticket is now accepted and I will adress it as soon as I am at my computer again.

**Comment by jens-maus** [comment]
**Originally by _johnblock@storeshop.com_ on 2013-01-02 11:22:43 +0100**

___
[http://www.mail2web.com]

Perhaps solves this (not noticed delivery issues) using

```
Sender: loggedinto@mailbox.net
From: "supplied@address.com" <eg:original@address.com>
```

Like Yam, before 2.8 removed the useful feature, you can put anything in the From: (except webmail addresses)

**Comment by jens-maus** [comment]
**Originally on 2013-01-29 00:32:54 +0100**

___
In (4bed163):
 * YAM_WR.c, mui/WriteWindow.c, misc: renamed the 'Bounce' mail functionality
  for redirecting emails to alternative recipients to 'Redirect' as that
  better reflects the purpose of this functionality and should not confuse
  users anymore. Also adapted the functionality to add all well-known
  Resent-#? headers to redirected emails. This should adapt YAM to be somehwat
  more RFC compliant. Furthermore, a Resent-CC and Resent-BCC can now be
  specified. In addition, YAM will now allow to bounce to multiple emails in
  one operation. That means, a user can now select more than one mail in the
  mail listview and when a redirect operation is performed YAM will redirect
  all of these mails in one run. This change should perfectly fix, fulfill and
  closes #362.

**Comment by jens-maus** [comment]
**Originally on 2013-07-01 21:31:19 +0200**

___
**Release Note:**
renamed the 'Bounce' mail functionality for redirecting emails to alternative recipients to 'Redirect' as that better reflects the purpose of this functionality and should not confuse users anymore. Also adapted the functionality to add all well-known Resent-#? headers to redirected emails. This should adapt YAM to be somehwat more RFC compliant. Furthermore, a Resent-CC and Resent-BCC can now be specified. In addition, YAM will now allow to bounce to multiple emails in one operation. That means, a user can now select more than one mail in the mail listview and when a redirect operation is performed YAM will redirect all of these mails in one run.
